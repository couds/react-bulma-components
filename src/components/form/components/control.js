import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


export default class Control extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    style: PropTypes.object,
    /**
     * the Html or React element you want to use to render this component
     */
    renderAs: PropTypes.string,
    /**
     * Is the button should be full width
     */
    fullwidth: PropTypes.bool,
    /**
     * True if this control will have an icon on the left
     */
    iconLeft: PropTypes.bool,
    /**
     * True if this control will have an icon on the Right
     */
    iconRight: PropTypes.bool,
    /**
     * Is the button should be displayed as loading
     */
    loading: PropTypes.bool,
    /**
     * The size of button using the classes from Bulma
     */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
  }

  static defaultProps = {
    children: null,
    className: '',
    style: {},
    renderAs: 'div',
    fullwidth: false,
    iconLeft: false,
    iconRight: false,
    loading: false,
    size: null,
  }

  render() {
    const {
      children,
      className,
      style,
      renderAs,
      fullwidth,
      iconLeft,
      iconRight,
      loading,
      size,
    } = this.props;
    const Element = renderAs;

    return (
      <Element
        style={style}
        className={classnames('control', className, {
          'is-expanded': fullwidth,
          'has-icons-left': iconLeft,
          'has-icons-right': iconRight,
          'is-loading': loading,
          [`is-${size}`]: size,
        })}
      >
        {children}
      </Element>
    );
  }
}
