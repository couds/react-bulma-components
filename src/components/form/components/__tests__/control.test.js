import React from 'react';
import { render } from 'react-testing-library';
import { Control } from '../control';

describe('Control component', () => {
  it('should render', () => {
    const { asFragment } = render(<Control />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Control className="other-class this-is-a-test">
        <p>Default</p>
      </Control>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Control style={{ height: 250 }}>
        <p>Default</p>
      </Control>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render as a html section element', () => {
    const { asFragment } = render(
      <Control renderAs="section">
        <p>Default</p>
      </Control>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be fullwidth, loading and size = large', () => {
    const { asFragment } = render(
      <Control fullwidth loading size="large">
        <p>Default</p>
      </Control>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
