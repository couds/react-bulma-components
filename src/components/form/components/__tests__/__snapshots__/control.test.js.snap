// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control component should be fullwidth, loading and size = large 1`] = `
<DocumentFragment>
  <div
    class="control is-expanded is-loading is-large"
  >
    <p>
      Default
    </p>
  </div>
</DocumentFragment>
`;

exports[`Control component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="control other-class this-is-a-test"
  >
    <p>
      Default
    </p>
  </div>
</DocumentFragment>
`;

exports[`Control component should render 1`] = `
<DocumentFragment>
  <div
    class="control"
  />
</DocumentFragment>
`;

exports[`Control component should render as a html section element 1`] = `
<DocumentFragment>
  <section
    class="control"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Control component should use inline styles 1`] = `
<DocumentFragment>
  <div
    class="control"
    style="height: 250px;"
  >
    <p>
      Default
    </p>
  </div>
</DocumentFragment>
`;
